src/Featurizers/Components/VectorNormsEstimator.h

```diff
@@ -0,0 +1,181 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "../../Traits.h"
+#include "NormUpdaters.h"
+#include "../Structs.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         VectorNormsEstimatorName("VectorNormsEstimator");
+
+namespace TypeSelector {
+
+// updater type selector is created to select different updater based on input parameter
+template <typename T, NormType NormT>
+struct UpdaterTypeSelector {

```

Comment:
>UpdaterTypeSelector [](start = 7, length = 19)

I've made changes in StatisticalMetricsEstimator to make it more usable within other Featurizers that are similar to these. Take a look at those changes and try to incorporate those concepts here.

Reply:
I don't think it applies here since function names such as reset, update and commit are the same for all three norm updaters and I just need to select appropriate updater type regarding NormType

Reply:
Resolved after discussion